### PR TITLE
Gh fix branchname error

### PR DESCRIPTION
**Pull request by ryantanjunming**
# Description

fix issue where branchname=jira-ticket 

# How Has This Been Tested?

# Checklist:
- [ ] I have updated the relevant component changelog(s)
- [x] This change does not affect any particular component (e.g. it's readme or docs change)

### CODE REVIEW

File: .github/workflows/synchronizing.yaml [hunk 1]
```diff
@@ -96,8 +96,8 @@ jobs:
 
       - name: apply changes to the files
         run: |
-          if echo "${{ github.event.head_commit.message }}" | grep -q "#"; then
-            branch_name=$(echo "${{ github.event.head_commit.message }}" | grep -o '[0-9]\+')
+          if echo "${{ github.event.head_commit.message }}" | grep -q '\(#\d*\)'; then
+            branch_name=$(echo "${{ github.event.head_commit.message }}" | grep -o '#\d*')
           else
```
Comment (frank-ang) [inline .github/workflows/synchronizing.yaml:101]: Whats an example of the undesirable jira merge-id that this pattern matches?
Suggest adding a comment to make it clear for future.

File: .github/workflows/synchronizing.yaml [hunk 2]
```diff
@@ -96,8 +96,8 @@ jobs:
 
       - name: apply changes to the files
         run: |
-          if echo "${{ github.event.head_commit.message }}" | grep -q "#"; then
-            branch_name=$(echo "${{ github.event.head_commit.message }}" | grep -o '[0-9]\+')
+          if echo "${{ github.event.head_commit.message }}" | grep -q '\(#\d*\)'; then
+            branch_name=$(echo "${{ github.event.head_commit.message }}" | grep -o '#\d*')
```
Comment (frank-ang) [inline .github/workflows/synchronizing.yaml:100]: '#\d+' ?

File: .github/workflows/synchronizing.yaml [hunk 3]
```diff
@@ -96,8 +96,8 @@ jobs:
 
       - name: apply changes to the files
         run: |
-          if echo "${{ github.event.head_commit.message }}" | grep -q "#"; then
-            branch_name=$(echo "${{ github.event.head_commit.message }}" | grep -o '[0-9]\+')
+          if echo "${{ github.event.head_commit.message }}" | grep -q '\(#\d*\)'; then
```
Comment (frank-ang) [inline .github/workflows/synchronizing.yaml:99]: Whats an example of the undesirable jira merge-id that this pattern matches?
Suggest adding a comment to make it clear for future.

Comment (ryantanjunming) [reply .github/workflows/synchronizing.yaml:99]: [CDO-123]

the merge ID will be something like (#23), so i'm just matching for the mergeID instead

File: .github/workflows/synchronizing.yaml (hunk 2 above)
Comment (ryantanjunming) [reply .github/workflows/synchronizing.yaml:100]: + doesnt work in this grep engine for some reason

File: .github/workflows/synchronizing.yaml (hunk 3 above)
Comment (frank-ang) [reply .github/workflows/synchronizing.yaml:99]: Regex should have # and at least 1 number?
Also '\d' is PCRE/exgended regex. Should use grep -e or -P.
Highly recommend a comment. Code intent is not self-evident.

File: .github/workflows/synchronizing.yaml (hunk 2 above)
Comment (frank-ang) [reply .github/workflows/synchronizing.yaml:100]: Is the intent here to set the branch name to e.g. "#123" ?

File: .github/workflows/synchronizing.yaml [hunk 4]
```diff
@@ -96,8 +96,8 @@ jobs:
 
       - name: apply changes to the files
         run: |
-          if echo "${{ github.event.head_commit.message }}" | grep -q "#"; then
-            branch_name=$(echo "${{ github.event.head_commit.message }}" | grep -o '[0-9]\+')
+          if echo "${{ github.event.head_commit.message }}" | grep -q '#\d*'; then
```
Comment (frank-ang) [inline .github/workflows/synchronizing.yaml:99]: For '\d' perl regex, should use ```grep -P``` or [0-9] character class. Tested on Alinux. Mac grep works different.

**Comment by frank-ang** [comment]
Discussion with @ryantanjunming . Testing was done in forked branch.

**Review by frank-ang — APPROVED** [review]
LGTM.

**Pull request merged**